### PR TITLE
Implement creating cluster Identity Providers

### PR DESCRIPTION
**Pull request by igoihman**

### CODE REVIEW

**Comment by igoihman** [comment]
@vkareh PTAL

File: cmd/ocm/create/idp/github.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+		}
+
+		registerURL, err := url.Parse(registerURLBase)
+		if err != nil {
+			return idpBuilder, fmt.Errorf("Error parsing URL: %v", err)
+		}
+
+		// Populate fields in the GitHub registration form
+		consoleURL := cluster.Console().URL()
+		oauthURL := strings.Replace(consoleURL, "console-openshift-console", "oauth-openshift", 1)
```
Comment (vkareh) [inline cmd/ocm/create/idp/github.go:79]: This line here will not work for RHMI clusters. It also needs to change `"solution-explorer"` to `"oauth-openshift"`

File: cmd/ocm/create/idp/github.go [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+	isInteractive := clientID == "" || clientSecret == "" || (organizations == "" && teams == "")
+
+	if isInteractive {
+		fmt.Println("To use GitHub as an identity provider, you must first register the application:")
+
+		if organizations == "" && teams == "" {
+			prompt := &survey.Input{
+				Message: "List of GitHub organizations or teams " +
+					"that will have access to this cluster",
```
Comment (vkareh) [inline cmd/ocm/create/idp/github.go:46]: This should have a ":" at the end:
```
$ ocm create idp -c 1ds0d3q8q40c0k3psfuuk3i7dpeedc8g
? Type of identity provider: github
To use GitHub as an identity provider, you must first register the application:
? List of GitHub organizations or teams that will have access to this cluster my-github-org
                                                                             ^---- missing colon
```

File: cmd/ocm/create/idp/github.go [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+		urlParams.Add("oauth_application[callback_url]", oauthURL+"/oauth2callback/"+idpName)
+
+		registerURL.RawQuery = urlParams.Encode()
+
+		fmt.Println("* Open the following URL:", registerURL.String())
+		fmt.Println("* Click on 'Register application'")
+
+		if clientID == "" {
+			prompt := &survey.Input{
+				Message: "Copy the Client ID provided by GitHub",
```
Comment (vkareh) [inline cmd/ocm/create/idp/github.go:92]: Missing ":" at the end

File: cmd/ocm/create/idp/github.go [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+			}
+			err = survey.AskOne(prompt, &clientID)
+			if err != nil {
+				return idpBuilder, errors.New("Expected a GitHub application Client ID")
+			}
+		}
+
+		if clientSecret == "" {
+			prompt := &survey.Input{
+				Message: "Copy the Client Secret provided by GitHub",
```
Comment (vkareh) [inline cmd/ocm/create/idp/github.go:102]: Missing ":" at the end

File: cmd/ocm/create/idp/google.go [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+		(args.mappingMethod != "lookup" && hostedDomain == "")
+
+	if isInteractive {
+		fmt.Println("To use Google as an identity provider, you must first register the application:")
+		instructionsURL := "https://console.developers.google.com/projectcreate"
+		fmt.Println("* Open the following URL:", instructionsURL)
+		fmt.Println("* Follow the instructions to register your application")
+
+		consoleURL := cluster.Console().URL()
+		oauthURL := strings.Replace(consoleURL, "console-openshift-console", "oauth-openshift", 1)
```
Comment (vkareh) [inline cmd/ocm/create/idp/google.go:43]: also `"solution-explorer"`

File: cmd/ocm/create/idp/openid.go [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+
+	if isInteractive {
+		fmt.Println("To use OpenID as an identity provider, you must first register the application:")
+		instructionsURL := "https://docs.openshift.com/dedicated/4/authentication/" +
+			"identity_providers/configuring-oidc-identity-provider.html"
+		fmt.Println("* Open the following URL:", instructionsURL)
+		fmt.Println("* Follow the instructions to register your application")
+
+		consoleURL := cluster.Console().URL()
+		oauthURL := strings.Replace(consoleURL, "console-openshift-console", "oauth-openshift", 1)
```
Comment (vkareh) [inline cmd/ocm/create/idp/openid.go:46]: also `"solution-explorer"`

**Comment by igoihman** [comment]
@vkareh I'm merging it as is, I will handle RHMI separately. Thanks for the review!

**Pull request merged**